bin/codemods/src/assign-then-export-react-class.js

```diff
@@ -0,0 +1,56 @@
+/**
+ * External dependencies
+ */
+const { get } = require( 'lodash' );
+
+/**
+ * Module variables
+ */
+const createClassIdentifier = {
+    type: 'CallExpression',
+    callee: {
+        type: 'MemberExpression',
+        object: {
+            name: 'React',
+        },
+        property: {
+            name: 'createClass',
+        }
+    },
+};
+
+const displayNamePropertyIdentifier = {
+    key: {
+        name: 'displayName',
+    },
+};
+
+const DEFAULT_EXPORT_NAME = '__CHANGE_ME__';
```

Comment:
I'm not sure if this is a terrible idea or not... by the nature of codemodding I feel that this might allow for cases where `__CHANGE_ME__` works it's way in to master. The obvious alternative is to skip the few components that don't have displayName properties but that would require more intervention than a quick variable name change.
Another idea I had was to infer the variable name from the file name but that seems like more effort than its worth, particularly to make it solid.

Open to ideas!

Reply:
Yeah, I'd say as a general rule, codemods should require zero manual intervention by design. Pretty much any codemod we've run on the entire codebase left things in a broken state simply because of edge cases that those codemods couln't handle -- requiring manually fixing (but as the result of their limited capabilities or a bug, not by design).

I'd much rather skip those components than having to remember that this particular codemod requires me to Search & Replace something manually after it's run. I think that's how most codemods work -- I kind of expect to check for skipped instances afterwards, find out why the codemod skipped it, fix it, and re-run the codemod.

Reply:
Yeah, I thought it seemed a bad idea :P I've actually just merged a PR that adds `displayName`s to all instances of `React.createClass` that were missing them, so there would naturally be fewer cases missed in the first place :)